forest/src/daemon.rs

```diff
@@ -68,7 +69,8 @@ pub(super) async fn start(config: Config) {
     let state_manager = Arc::new(StateManager::new(Arc::clone(&db)));
 
     // Initialize mpool
-    let subscriber = chain_store.subscribe().await;
+    let publisher = chain_store.publisher_arc();
+    let subscriber = (*publisher.write().await).subscribe();
```

Comment:
```suggestion
    let subscriber = publisher.write().await.subscribe();
```
This will deref automatically